awx_collection/plugins/modules/tower_project.py

```diff
@@ -133,107 +148,128 @@
-
-                if (scm_update_cache_timeout is not None) and (scm_update_on_launch is not True):
-                    module.warn('scm_update_cache_timeout will be ignored since scm_update_on_launch was not set to true')
-
-                result = project.modify(name=name, description=description,
-                                        organization=org['id'],
-                                        scm_type=scm_type, scm_url=scm_url, local_path=local_path,
-                                        scm_branch=scm_branch, scm_clean=scm_clean, credential=scm_credential,
-                                        scm_delete_on_update=scm_delete_on_update,
-                                        scm_update_on_launch=scm_update_on_launch,
-                                        scm_update_cache_timeout=scm_update_cache_timeout,
-                                        job_timeout=job_timeout,
-                                        custom_virtualenv=custom_virtualenv,
-                                        create_on_missing=True)
-                json_output['id'] = result['id']
-                if wait and scm_type != '':
-                    project.wait(pk=None, parent_pk=result['id'])
-            elif state == 'absent':
-                result = project.delete(name=name)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            module.fail_json(msg='Failed to update project: {0}'.format(excinfo), changed=False)
-
-    json_output['changed'] = result['changed']
-    module.exit_json(**json_output)
+    # Attempt to look up the related items the user specified (these will fail the module if not found)
+    org_id = module.resolve_name_to_id('organizations', organization)
+    if scm_credential is not None:
+        scm_credential_id = module.resolve_name_to_id('credentials', scm_credential)
+
+    # Attempt to look up project based on the provided name and org ID
+    project = module.get_one('projects', **{
+        'data': {
+            'name': name,
+            'organization': org_id
+        }
+    })
+
+    # Create the data that gets sent for create and update
+    project_fields = {
+        'name': name,
+        'description': description,
+        'scm_type': scm_type,
+        'scm_url': scm_url,
+        'scm_branch': scm_branch,
+        'scm_refspec': scm_refspec,
+        'scm_clean': scm_clean,
+        'scm_delete_on_update': scm_delete_on_update,
+        'timeout': job_timeout,
+        'organization': org_id,
+        'scm_update_on_launch': scm_update_on_launch,
+        'scm_update_cache_timeout': scm_update_cache_timeout,
+        'custom_virtualenv': custom_virtualenv,
+    }
+    if scm_credential is not None:
+        project_fields['credential'] = scm_credential_id
+    if scm_allow_override is not None:
+        project_fields['scm_allow_override'] = scm_allow_override
```

Comment:
Why are some of these going through the `is not None` check but others aren't?

Reply:
I've updated the other non-required params in the modules to go through the same check; turns out this affected idempotency behavior.